test/test_dynamic_shapes.py

```diff
@@ -7,6 +7,18 @@
 
 class TestDynamicShapes(unittest.TestCase):
 
+  def test_simple_expand(self):
+    size1 = 5
+    size2 = 2
+    t1 = torch.zeros([size1, size2], device=dev)
+    t1[3][0] = 1
+    # t2 has size [<=10, 2]
+    t2 = torch.nonzero(t1)
+    t5 = torch.ones(1, device=dev)
+    t5.expand(t2.size(0))
```

Comment:
I felt like we should add a check here to test `t5.shape()`, this can be down in a follow up pr.

Reply:
I can add a test later.